### PR TITLE
Modularize Bazel build

### PR DESCRIPTION
**Pull request by mofeing**
Following EnzymeAD/Reactant.jl#421, I'm moving build deps to different files inside the `third_party` folder.

I would prefer to also pack the calls in `WORKSPACE` because it's similar in ReactantExtra, but we can do that later.

### CODE REVIEW

**Comment by mofeing** [comment]
@wsmoses the tests pass. the failing tests are not due to this PR and are also failing on main.

File: third_party/hedron_compile_commands/workspace.bzl [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def repo():
+    http_archive(
+        name = "hedron_compile_commands",
+
+        # Replace the commit hash (0e990032f3c5a866e72615cf67e5ce22186dcb97) in both places (below) with the latest (https://github.com/hedronvision/bazel-compile-commands-extractor/commits/main), rather than using the stale one here.
+        # Even better, set up Renovate and let it do the work for you (see "Suggestion: Updates" in the README).
+        url = "https://github.com/hedronvision/bazel-compile-commands-extractor/archive/4f28899228fb3ad0126897876f147ca15026151e.tar.gz",
```
Comment (wsmoses) [inline third_party/hedron_compile_commands/workspace.bzl:12]: can you put this commit / sha inside of workspace.bzl ?

**Review by wsmoses — APPROVED** [review]
small comment otherwise lgtm!

File: third_party/hedron_compile_commands/workspace.bzl (hunk 1 above)
Comment (mofeing) [reply third_party/hedron_compile_commands/workspace.bzl:12]: sure!

**Pull request merged**